### ISSUE TITLE
configure_spark_with_delta_pip overwrites "spark.jars.packages"

### ISSUE THREAD

**Issue by martinitus** [delta-io/delta#889]
I just tried configuring a unittest with both kafka and delta lake settings and notices that my manually specified 

` .config('spark.jars.packages', 'org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0')` gets overwritten by `configure_spark_with_delta_pip`. I think a more reasonable action would be to append to the list instead of replacing.

alternatively, one could just provide a helper:

```python
def delta_jars()->List[str]:
    """
    return the list of jars that need to be added to the spark session to use delta lake.
   Add these jars to your pyspark session setup. E.g.
    builder.config("spark.jars.packages",",".join(delta_jars() + ["my-jars"] ))
    """
    import importlib_metadata  # load this library only when this function is called
    try:
        delta_version = importlib_metadata.version("delta_spark")
    except Exception as e:
        msg = '''
This function can be used only when Delta Lake has been locally installed with pip.
See the online documentation for the correct usage of this function.
        '''
        raise Exception(msg) from e

    scala_version = "2.12"
    return [ f"io.delta:delta-core_{scala_version}:{delta_version}"]
```

**Comment by zsxwing** [comment]
Thanks for reporting the issue. Yep. This is a bug but currently SparkSession Builder doesn't provide an API to append a value to an existing config (or at least expose an API to get the existing configs) 😞  

A better solution would be asking Spark to provide such API.

Instead introducing a new API, maybe just allow users to pass their packages? Such as
```
def configure_spark_with_delta_pip(
    spark_session_builder: SparkSession.Builder,
    extraPackages: List[str] = []
)
```

**Comment by zsxwing** [comment]
Fixed by https://github.com/delta-io/delta/commit/6e1a488f3e0ca3ea4e605948aa3870c25d0b25ee

**Comment by cgrisham** [comment]
Does this fix only apply to 1.2.* releases? Using 1.0.1 to match my Spark 3.1.2 environment and it doesn't recognize the extra_package argument. Any solution besides updating my spark installation?

**Comment by zsxwing** [comment]
@cgrisham Right. This is only available in 1.2.*. If you use pyspark `--packages` such as `pyspark --packages io.delta:delta-core_2.12:1.0.1 --conf "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension" --conf "spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog"`, you don't need to call `configure_spark_with_delta_pip`.
